pkg/minikube/minikube.go

```diff
@@ -0,0 +1,174 @@
+// limitations under the License.
+
+package minikube
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"knative.dev/kn-plugin-quickstart/pkg/install"
+)
+
+var clusterName = "minikube-knative"
+var minikubeVersion = 1.23
+
+// SetUp creates a local Minikube cluster and installs all the relevant Knative components
+func SetUp() error {
+	if err := createMinikubeCluster(); err != nil {
+		return fmt.Errorf("creating cluster: %w", err)
+	}
+	if err := install.Serving(); err != nil {
+		return fmt.Errorf("install serving: %w", err)
+	}
+	if err := install.Kourier(); err != nil {
+		return fmt.Errorf("install kourier: %w", err)
+	}
+	if err := install.KourierMinikube(); err != nil {
+		return fmt.Errorf("configure kourier: %w", err)
+	}
+	if err := install.Eventing(); err != nil {
+		return fmt.Errorf("install eventing: %w", err)
+	}
+	return nil
+}
+
+func createMinikubeCluster() error {
+	if err := checkMinikubeVersion(); err != nil {
+		return fmt.Errorf("minikube version: %w", err)
+	}
+	if err := checkForExistingCluster(); err != nil {
+		return fmt.Errorf("existing cluster: %w", err)
+	}
+	return nil
+}
+
+// checkMinikubeVersion validates that the user has the correct version of Minikube installed.
+// If not, it prompts the user to download a newer version before continuing.
+func checkMinikubeVersion() error {
+	versionCheck := exec.Command("minikube", "version", "--short")
+	out, err := versionCheck.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("minikube version: %w", err)
+	}
+	fmt.Printf("Minikube version is: %s\n", string(out))
+
+	userMinikubeVersion, err := parseMinikubeVersion(string(out))
+	if err != nil {
+		return fmt.Errorf("parsing minikube version: %w", err)
+	}
+	if userMinikubeVersion < minikubeVersion {
+		var resp string
+		fmt.Printf("WARNING: Please make sure you are using Minikube version >= %.2f\n", minikubeVersion)
+		fmt.Println("Download from https://github.com/kubernetes/minikube/releases/")
+		fmt.Print("Do you want to continue at your own risk [Y/n]: ")
+		fmt.Scanf("%s", &resp)
+		if resp == "n" || resp == "N" {
+			fmt.Println("Installation stopped. Please upgrade minikube and run again")
+			os.Exit(0)
+		}
+	}
+
+	return nil
+}
+
+// checkForExistingCluster checks if the user already has a Minikube cluster. If so, it provides
+// the option of deleting the existing cluster and recreating it. If not, it proceeds to
+// creating a new cluster
+func checkForExistingCluster() error {
+
+	getClusters := exec.Command("minikube", "profile", "list")
+	out, err := getClusters.CombinedOutput()
+	if err != nil {
+		// there are no existing minikube profiles, the listing profiles command will error
+		// if there were no profiles, we simply want to create a new one and not stop the install
+		// so if the error contains a "no profile found" string, we ignore it and continue onwards
+		if !strings.Contains(string(out), "No minikube profile was found") {
```

Comment:
The error code we'd get here is `14`, but that's a [general `ExProgramUsage`](https://minikube.sigs.k8s.io/docs/contrib/errorcodes/#error-codes-specific-to-the-minikube-program) error which would also be thrown by, for example, a typo in the minikube command:

```
$ minikube profil list
Error: unknown command "profil" for "minikube"

Did you mean this?
        profile

Run 'minikube --help' for usage.

$ echo $?
14

$ minikube profile list 

🤹  Exiting due to MK_USAGE_NO_PROFILE: No minikube profile was found. 
💡  Suggestion: 

    You can create one using 'minikube start'.
    


$ echo $?
14

```

Instead of using seaching for "No minikube profile was found", maybe it would be better to use `MK_USAGE_NO_PROFILE`, which seems to the be actual error?

Reply:
what if you add `-o json` to the arguments?

Reply:
That provides a list of profiles... is checking if the `valid` list is empty much better than just parsing the string?

```
$ minikube profile list -o json
{"invalid":[],"valid":[]}
```